sw/device/examples/sram_program/BUILD

```diff
@@ -42,23 +42,55 @@ opentitan_gdb_fpga_cw310_test(
     exit_success_pattern = "sram_program\\.c:47\\] PC: 0x1000208c, SRAM: \\[0x10000000, 0x10020000\\)",
     gdb_script = """
         target extended-remote :3333
-        set pagination off
 
         echo :::: Send OpenOCD the 'reset halt' command.\\n
         monitor reset halt
 
-        echo :::: Set RAM as RWX by executing code between _start and test_rom_main()\\n
-        file test_rom.elf
-        break rom_test_main
+        # Disable the watchdog timer as quickly as possible. If it fires, the
+        # exception will steal control from the debugger. If we're unlucky, the
+        # watchdog timer might fire before the commands below have disabled it,
+        # causing this test to fail. The proper solution would be to disable ROM
+        # execution by zeroing out CREATOR_SW_CFG_ROM_EXEC_EN in the OTP.
+        file rom.elf
+        break sec_mmio_init
         continue
+        finish
+        print watchdog_disable()
         delete breakpoints
 
+        # Before transferring the SRAM program to the device, we must configure
+        # the PMP unit to enable writing to and executing from SRAM. Due to
+        # implementation details of OpenTitan's hardware debug module, we cannot
+        # set pmpcfg* registers to arbitrary values[1]. However, we can safely
+        # modify unused PMP configuration registers. Thankfully, pmp0cfg (the
+        # lowest byte of CSR pmpcfg0) is unused and has the highest priority.
+        #
+        # In more detail, the problem is that our debug module implements the
+        # "Access Register" abstract command by assembling instructions in the
+        # program buffer and then executing the buffer. If one of those
+        # instructions clobbers the PMP configuration register that allows
+        # execution from the program buffer, subsequent instruction fetches will
+        # generate exceptions. To work around this, we simply leave any in-use
```

Comment:
Ah, makes sense. Just out of curiosity, can we touch those registers after configuring `pmpcfg0` or no?

Reply:
Yeah, after setting up `pmpcfg0` to cover everything, we can go ahead and clobber the other registers. It just wasn't necessary, so I didn't do it :)

Specifically, we *could* do this right after setting `pmpaddr0`.

```
        monitor reg pmpcfg0 0x0000009f
        monitor reg pmpcfg1 0x0
        monitor reg pmpcfg2 0x0
        monitor reg pmpcfg3 0x0
```

Reply:
(To be clear, I did test that.)